### ISSUE TITLE
Which options do we have to collect email addresses of users?

### ISSUE THREAD

**Issue by mariobehling** [fossasia/fossasia-companion-android#68]
Do we get the email addresses of user, if we implement Google+ sign on? What other options do we have?

**Comment by ty221** [comment]
If somebody logs in using FB, we do get it. If somebody uses G+, we probably can fetch it - https://developers.google.com/+/mobile/android/people

**Comment by mananwason** [comment]
Yes @ty221 @mariobehling  we can fetch it very easily !!

**Comment by mariobehling** [comment]
@ty221 from FB we just get the FB email adress which is often just the facebook email, right? About G+ what are the steps to implement this? How is the experience for the user?
a) They download the app
b) They sign in with their Google-ID
c) They use the app
- once we have their email address, we don't want them to fill in their info again every time.
- is it possible to get their email without having them signing in?
- how/where do we store user info? what is a simple way?

**Comment by mananwason** [comment]
@mariobehling We can use the primary email account used to setup the device by the user if we don't want them to sign in but it's not really much advisable. This is done by Accountmanager(API 5+)

**Comment by batraabhishek** [comment]
> once we have their email address, we don't want them to fill in their info again every time.

Yes, Google+ login will provide the email also. Once the user has logged in, we can store the user's name and email in shared preference (App storage). This data is not volatile and retained even if the app is restarted. We will have to show a sign out button (may be in navigation drawer). One the user clicks "Sign Out", we will delete the data from shared preference. 

> is it possible to get their email without having them signing in?

Hmm. Even if we do so we will have to ask the user to provide an email address (Just like subscribe button on websites), which may not be always correct. Or we can follow @mananwason idea, but that comes with a lot of drawbacks.

> how/where do we store user info? what is a simple way?

The simplest way is to have a database hosted somewhere, But we can explore other options as well.

**Comment by mariobehling** [comment]
> The simplest way is to have a database hosted somewhere, But we can explore other options as well.

Ok, great. What kind of DB. We have a test server here.

**Comment by batraabhishek** [comment]
Any Database would do. We can just write a simple REST Api, which takes the params via POST. This can be done with NodeJS in few seconds ;)

Another approach is storing directly in google sreadsheet. It is also possible :)

**Comment by mariobehling** [comment]
In this case for the moment it would be great to do it in a Google spreadsheet. Each sheet can be another event.

**Comment by Nitish18** [comment]
@mariobehling sir, can I work on this issue ??

**Comment by dalmia** [comment]
Hi everyone! I would like to work on this issue.
As mentioned by @mananwason we can get it without having the user signed in because users tend to back away when they have to login, the dropout rates increase in this step. So, just taking the primary email account seems the best fit, if only email is required.
Also, the data can be easily stored as suggested by @batraabhishek  and  also can be stored locally in a SQLite Database.
